In [1]:
import pandas as pd
from zipfile import ZipFile
import yfinance as fp
import datetime

In [2]:
# with ZipFile("Data/NIFTY50_all.csv.zip","r") as ref:
#     unzip = ref.extractall("Data/Bnk_NIFTY50")
# ref.close()

In [3]:
today = datetime.date.today()
year  = today-datetime.timedelta(days=7300) #20 Years
Stocks = ['AAPL', 'MSFT', 'IBM', 'GE', 'JNJ']
data = fp.download(Stocks,start=year,end=today,group_by="ticker")
data

C:\Users\lipun\AppData\Local\Temp\ipykernel_24452\1640078772.py:4: FutureWarning: YF.download() has changed argument auto_adjust default to True
  data = fp.download(Stocks,start=year,end=today,group_by="ticker")
[*********************100%***********************]  5 of 5 completed


Ticker            MSFT                                                 \
Price             Open        High         Low       Close     Volume   
Date                                                                    
2005-07-21   18.294991   18.420204   18.086303   18.392380  112932100   
2005-07-22   18.079358   18.322828   17.828932   17.863714   97558900   
2005-07-25   17.870661   18.016742   17.842835   17.870661   45174600   
2005-07-26   17.891533   17.905446   17.759365   17.766321   51476400   
2005-07-27   17.815009   17.947177   17.759359   17.891527   57977300   
...                ...         ...         ...         ...        ...   
2025-07-09  500.299988  506.779999  499.739990  503.510010   18659500   
2025-07-10  503.049988  504.440002  497.750000  501.480011   16492100   
2025-07-11  498.470001  505.029999  497.799988  503.320007   16459500   
2025-07-14  501.519989  503.970001  501.029999  503.019989   12058800   
2025-07-15  503.019989  508.299988  502.790009  505.820007   14906100   

Ticker            AAPL                                                 ...  \
Price             Open        High         Low       Close     Volume  ...   
Date                                                                   ...   
2005-07-21    1.313367    1.323586    1.289324    1.301045  404264000  ...   
2005-07-22    1.305554    1.322385    1.304051    1.322385  301106400  ...   
2005-07-25    1.322084    1.330800    1.314270    1.316674  294627200  ...   
2005-07-26    1.322685    1.325690    1.303149    1.311264  268592800  ...   
2005-07-27    1.317275    1.324488    1.282412    1.322083  283749200  ...   
...                ...         ...         ...         ...        ...  ...   
2025-07-09  209.529999  211.330002  207.220001  211.139999   48749400  ...   
2025-07-10  210.509995  213.479996  210.029999  212.410004   44443600  ...   
2025-07-11  210.570007  212.130005  209.860001  211.160004   39765800  ...   
2025-07-14  209.929993  210.910004  207.539993  208.619995   38840100  ...   
2025-07-15  209.220001  211.889999  208.919998  209.110001   42225300  ...   

Ticker             IBM                                               \
Price             Open        High         Low       Close   Volume   
Date                                                                  
2005-07-21   43.973132   44.155055   43.349401   43.869179  8645504   
2005-07-22   43.650847   43.988702   43.588474   43.889946  4780638   
2005-07-25   43.650861   43.931544   43.619676   43.765213  4662859   
2005-07-26   43.739218   43.843171   43.583282   43.723621  4751873   
2005-07-27   43.557322   43.728846   43.042742   43.593704  5085861   
...                ...         ...         ...         ...      ...   
2025-07-09  291.390015  291.600006  288.630005  290.140015  2971300   
2025-07-10  288.899994  288.899994  282.209991  287.429993  3487800   
2025-07-11  285.010010  287.429993  282.920013  283.589996  3790700   
2025-07-14  282.829987  284.929993  281.709991  283.790009  2857100   
2025-07-15  283.769989  284.149994  280.730011  282.700012  2862600   

Ticker              GE                                               
Price             Open        High         Low       Close   Volume  
Date                                                                 
2005-07-21  102.658444  103.183701  102.133186  102.133186  3399293  
2005-07-22  102.833561  102.833561  101.753863  102.337486  3133979  
2005-07-25  102.716869  102.716869  101.257819  101.462082  3783503  
2005-07-26  101.637149  102.454218  101.228614  101.257797  3420410  
2005-07-27  101.287000  101.607986  101.111911  101.549629  3482466  
...                ...         ...         ...         ...      ...  
2025-07-09  248.990005  251.720001  246.820007  250.330002  5589500  
2025-07-10  250.610001  252.679993  246.750000  252.570007  5462500  
2025-07-11  251.500000  256.880005  251.399994  255.419998  7867200  
2025-07-14  256.190002  262.459991  256.000000  2

In [4]:
new = []
for i in Stocks:
    df = data[i].copy()
    df = df.reset_index()
    df["Stocks"] = i
    new.append(df)
new_data = pd.concat(new)
new_data  

Price       Date        Open        High         Low       Close     Volume  \
0     2005-07-21    1.313367    1.323586    1.289324    1.301045  404264000   
1     2005-07-22    1.305554    1.322385    1.304051    1.322385  301106400   
2     2005-07-25    1.322084    1.330800    1.314270    1.316674  294627200   
3     2005-07-26    1.322685    1.325690    1.303149    1.311264  268592800   
4     2005-07-27    1.317275    1.324488    1.282412    1.322083  283749200   
...          ...         ...         ...         ...         ...        ...   
5022  2025-07-09  155.789993  156.690002  155.070007  156.279999    6147400   
5023  2025-07-10  156.100006  159.240005  155.820007  157.690002    6732900   
5024  2025-07-11  157.089996  157.179993  155.179993  156.899994    7872000   
5025  2025-07-14  156.869995  157.470001  155.520004  156.820007   10185600   
5026  2025-07-15  156.360001  157.190002  154.800003  155.169998    6864200   

Price Stocks  
0       AAPL  
1       AAPL  
2       AAPL  
3       AAPL  
4       AAPL  
...      ...  
5022     JNJ  
5023     JNJ  
5024     JNJ  
5025     JNJ  
5026     JNJ  

[25135 rows x 7 columns]

In [5]:
new_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 25135 entries, 0 to 5026
Data columns (total 7 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   Date    25135 non-null  datetime64[ns]
 1   Open    25135 non-null  float64       
 2   High    25135 non-null  float64       
 3   Low     25135 non-null  float64       
 4   Close   25135 non-null  float64       
 5   Volume  25135 non-null  int64         
 6   Stocks  25135 non-null  object        
dtypes: datetime64[ns](1), float64(4), int64(1), object(1)
memory usage: 1.5+ MB


In [6]:
#Saving File
new_data.to_csv("Data/DataPreparation.csv",index=False)

<h1><b>FINAL REPORT: DATA-PREPARATION</b></h1>
<li>
    Data-Source: <a href="yfinance">Yfinance</a>
</li>
<li>Converted Raw Dataset to Structured Format With a New Feature as Stocks</li>
<li><b>Total Samples:</b> 25135</li>
<li><b>Total Independent Features: </b>"Date","Open","High","Low","Volume" and "Stocks"</li>
<li><b>Dependent Feature:</b> "Close"</li>
<center><b><li>Stocks Names</li></b></center>
<table>
    <th>Stocks</th>
    <th>Symbols</th>
    <tr><td>AAPL</td><td>Apple</td></tr>
    <tr><td>MSFT</td><td>Microsoft</td></tr>
    <tr><td>IBM</td><td>IBM</td></tr>
    <tr><td>GE</td><td>General Electric</td></tr>
    <tr><td>JNJ</td><td>Johnson & Johnson</td></tr>
</table>
<center><li><b>Feature Dtypes</b></li></center>
<table>
    <th>Features</th>
    <th>DataTypes</th>
    <tr><td>Date</td><td>datetime64</td></tr>
    <tr><td>Open</td><td>float64</td></tr>
    <tr><td>High</td><td>float64</td></tr>
    <tr><td>Low</td><td>float64</td></tr>
    <tr><td>Close</td><td>float64</td></tr>
    <tr><td>Volume</td><td>int64</td></tr>
    <tr><td>Stocks</td><td>object</td></tr>
</table>

